framework/validators/RequiredValidator.php

```diff
@@ -88,6 +88,10 @@ protected function validateValue($value)
      */
     public function clientValidateAttribute($model, $attribute, $view)
     {
+        if ($this->whenClient === null && $this->when !== null && !call_user_func($this->when, $model, $attribute)) {
```

Comment:
Doesn't look right to me. It could depend on the value of the model attribute which we don't have till client script validation passes.